keeper/valuation_engine.go

```diff
@@ -41,7 +44,11 @@ import (
 // Returns
 //   - (num, den) as math.Int, suitable for computing: floor(x * num / den).
 func (k Keeper) UnitPriceFraction(ctx sdk.Context, srcDenom, underlyingAsset string) (num, den math.Int, err error) {
-	if srcDenom == underlyingAsset {
+	// Currently, we are treating "uylds.fcc" as a universal stablecoin equivalent to the underlying asset.
+	// This is a temporary measure until we have a more robust multi-currency support and stablecoin handling.
+	// The assumption is that "uylds.fcc" is always pegged 1:1 with the underlying asset for vault valuation purposes.
+	// For more information, see https://github.com/ProvLabs/vault/issues/73
+	if srcDenom == underlyingAsset || underlyingAsset == "uylds.fcc" {
```

Comment:
For this specific case ... how will this resolve when we have a vault with a `recieptToken` and `uylds.fcc` where the receipt token is intended to have a 1:1 value relationship for the YLDS that was used for investment?